### ISSUE TITLE
User experience outside of RStudio IDE

### ISSUE THREAD

**Issue by cderv** [rstudio/blogdown#600]
This ticket is to compile different behavior that should be improve to use **blogdown** outside of the RStudio IDE (in vscode for example). 

Solution may not to be implemented in **blogdown** directly, but I believe listing the known limitation can help fix them in the right place. 

- [ ] When using in VSCODE, files are open in default notepad and not in vscode IDE. 
    * `file.edit()` is used when **rstudioapi** is not available.
    * **rstudioapi** is supported in vscode but not by default: https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r#rstudio-addin-support - 
    * `navigateToFile()` seems to be supported: https://github.com/Ikuyadeu/vscode-R/wiki/RStudio-addin-support
    * Maybe activating it can work ?

**Comment by yihui** [comment]
I don't have VS Code installed. It sounds like `options(vsc.rstudioapi = TRUE)` should just work? Because `file.edit()` won't be called if `rstudioapi::navigateToFile()` has succeeded:
https://github.com/rstudio/blogdown/blob/f324246c8a5ae72b9cd7c6d5b2eac3b65187652a/R/utils.R#L566-L568

**Comment by cderv** [comment]
Yes I think so. I did not try yet.

**Comment by yihui** [comment]
@milesmcbain Could you do us a favor to quickly test if `blogdown:::open_file()` can open a file in VS Code? Thanks!

**Comment by MilesMcBain** [comment]
Yes seems to be working (opening in new VSCode editor) with API emulation enabled.  👍

**Comment by MilesMcBain** [comment]
Also note that when I developed the emulation `blogdown` was one of my test cases. It seems that all the addins worked, with some unusual theme behaviour on the preview, although that may be fixed by recent stuff in VSCode.

**Comment by yihui** [comment]
Good to know! Thanks!

**Comment by cderv** [comment]
Does this means it would be interesting for us to set `options(vsc.rstudioapi = TRUE)` in `blogdown::open_file()` so that the file always open in the IDE and not outside ? Or would that be too much ? 

Also @MilesMcBain I did not manage to make it work, I may be missing something in my R config with vscode. 😞 I'll keep trying now I know it is working.

**Comment by yihui** [comment]
> Does this means it would be interesting for us to set `options(vsc.rstudioapi = TRUE)` in `blogdown::open_file()` so that the file always open in the IDE and not outside ?

Sounds like a good idea. Done with 3a67fe6643c18d58c5647f646c22f573f0b578df. However, I have no idea how the rstudioapi emulation in VS Code works. In particular, I'm wondering if it would be too late to set the option `vsc.rstudioapi` inside `open_file()`. In other words, does the option have to be set in `.Rprofile`?

**Comment by cderv** [comment]
> In other words, does the option have to be set in .Rprofile?

Good question. Doc says `.Rprofile`: https://github.com/Ikuyadeu/vscode-R/wiki/RStudio-addin-support#enabling-rstudio-addin-support
@MilesMcBain, maybe you know for sure ?

**Comment by MilesMcBain** [comment]
> Also @MilesMcBain I did not manage to make it work, I may be missing something in my R config with vscode. 😞 I'll keep trying now I know it is working.

The extension needs to have attached a terminal using one of its commands: "R: Create R Terminal", or "R: Attach Active Terminal". (See command palette). As part of this it runs an `init.R` script that sets up a bunch of infrastructure including the `rstudioapi` emulation if enabled. Since your simple case isn't working it makes me think this has not happened. Maybe you just ran the R binary from the built-in terminal without attaching it to the extension?

This also partly answers the second question:

> In other words, does the option have to be set in .Rprofile?

Yes. 

One further thing I would add is that: last time I checked the plan was for this option to default to `TRUE` in some future release. The main reason it exists is because when we first released the emulation there was some unexpected platform specific behaviour buried in some RStudio libraries. I think we've probably let all that settle out by now.
